src/go/k8s/controllers/redpanda/redpanda_controller_utils.go

```diff
@@ -1,21 +1,95 @@
 package redpanda
 
 import (
+	"context"
+	"fmt"
 	"io"
 	"net"
 	"net/http"
 	"os"
 	"path/filepath"
 	"time"
 
-	"k8s.io/utils/pointer"
-
+	"github.com/fluxcd/pkg/runtime/logger"
 	"github.com/fluxcd/source-controller/controllers"
 	"github.com/go-logr/logr"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/cli"
 	"helm.sh/helm/v3/pkg/registry"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	K8sInstanceLabelKey  = "app.kubernetes.io/instance"
+	K8sNameLabelKey      = "app.kubernetes.io/name"
+	K8sComponentLabelKey = "app.kubernetes.io/component"
+	K8sManagedByLabelKey = "app.kubernetes.io/managed-by"
+
+	EnvHelmReleaseNameKey = "REDPANDA_HELM_RELEASE_NAME"
 )
 
+var UpdateEventFilter = predicate.Funcs{
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	UpdateFunc:  func(e event.UpdateEvent) bool { return true },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+}
+
+var DeleteEventFilter = predicate.Funcs{
+	CreateFunc:  func(e event.CreateEvent) bool { return false },
+	UpdateFunc:  func(e event.UpdateEvent) bool { return false },
+	DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+	GenericFunc: func(e event.GenericEvent) bool { return false },
+}
+
+// TODO check assumptions here
+// Check to see if the release name of a helm chart matches the name of a redpanda object
+// this is by design for the operator
+func isValidReleaseName(releaseName string, redpandaNameList []string) bool {
+	for i := range redpandaNameList {
+		if releaseName == redpandaNameList[i] {
+			return true
+		}
+	}
+	return false
+}
+
+func getHelmValues(releaseName, namespace string) (map[string]interface{}, error) {
+	settings := cli.New()
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(settings.RESTClientGetter(), namespace, os.Getenv("HELM_DRIVER"), func(format string, v ...interface{}) { fmt.Println(format, v) }); err != nil {
```

Comment:
I wonder why `getHelmValues` function doesn't pass structured logger to `actionConfig` initialization? 

The usage of `fmt.Println(fromat, v)` seems to me as wrong usage. If operator operator uses different schemes to produce it's logs, then it could be potentially dropped from ingesting into the central logging system

Reply:
It is an afterthought, but yea we can definitely pass something else to it, i will take a look.